hooks/lib/check-boilerplate.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euo pipefail
+
+HOOKS_LIB="$(dirname "$(test -L "$0" && echo "$(dirname $0)/$(readlink "$0")" || echo "$0")")"
+
+for FILE in "$@"; do
+  # Check for files without the required boilerplate.
+  EXT=${FILE##*.}
+  REF_FILE="${HOOKS_LIB}/boilerplate.${EXT}.txt"
+
+  if [ ! -e $REF_FILE ]; then
+    continue
+  fi
+
+  LINES=$(cat "${REF_FILE}" | wc -l | tr -d ' ')
+  if [[ "${EXT}" == "py" && -x "${FILE}" ]]; then
+    # remove shabang and blank line from top of executable python files.
+    HEADER=$(cat "${FILE}" | tail --lines=+3 | head "-${LINES}")
+  else
+    HEADER=$(head "-${LINES}" "${FILE}")
+  fi
+
+  # Treat errors from diff other than the usual "files differ"
+  # code 1 as fatal problems.
+  DIFFER=$(echo "${HEADER}" | diff - "${REF_FILE}" || [[ $? -le 1 ]])
```

Comment:
FYI, the content of this file is mostly just copied from the previous `boilerplate.sh` file, replacing "echo 0" with "continue". I added the diff exit code check here since the original wasn't using strict `-euo pipefail` error checking.